templates/common/_base/units/machine-config-daemon-pull.service

```diff
@@ -12,11 +12,12 @@ contents: |
   Before=machine-config-daemon-host.service
 
   [Service]
-  # Need oneshot to delay kubelet
-  Type=oneshot
+  Type=simple
   ExecStart=/usr/bin/sh -c "/usr/bin/podman run --authfile=/var/lib/kubelet/config.json --quiet --net=host --entrypoint=cat '{{ .Images.setupEtcdEnvKey }}' /usr/bin/machine-config-daemon > /usr/local/bin/machine-config-daemon"
   ExecStart=/usr/bin/chmod +x /usr/local/bin/machine-config-daemon
   ExecStart=/usr/sbin/restorecon /usr/local/bin/machine-config-daemon
+  Restart=on-failure
+  RestartSec=30
```

Comment:
Seems reasonable, though at some point we will need to bubble up provisioning failures.